### PR TITLE
Add dropdown in UI to select issue type to be displayed.

### PR DESCRIPTION
**Pull request by suryalijo**
1)Added a drop-down in UI to select the category of issue to be displayed. The options in dropdown include All issues and the 5 main categories. Data is filtered after checking the selected category and the status of the "Hide manual Check" check box.
2)Moved "Hide manual Check" button and checkbox to the right.

### CODE REVIEW

File: src/components/TopIssues.js [hunk 1]
```diff
@@ -24,6 +26,23 @@ class TopIssues extends React.Component {
...
+        let issue= e.target.value;      
+        if(issue=="all-issues"){
+            this.setState({            
+                filteredData: (!this.state.showManual ? this.state.data.filter(item => item.manual === false) : this.state.data) ,
+                selectedIssue: issue          
+            });                   
+        }
+        else{
+            this.setState({            
+                filteredData: (this.state.showManual ? this.state.data.filter(item => item.category === issue) : this.state.data.filter(item => item.category === issue && item.manual === false)),
```
Comment (Winerip) [inline src/components/TopIssues.js:40]: nit: This line is a bit long. would read better if you broke the ternary into separate lines.

File: src/components/TopIssues.js [hunk 2]
```diff
@@ -36,11 +55,20 @@ class TopIssues extends React.Component {
     }
     handleCheckChange(e) {
         e.stopPropagation();
-        this.setState({
-            filteredData: (this.state.showManual ? this.state.data.filter(item => item.manual === false) : this.state.data),
-            showManual: !this.state.showManual
-        });
+        if(this.state.selectedIssue=="all-issues"){
```
Comment (Winerip) [inline src/components/TopIssues.js:58]: Looks like there's a lot of repeated code - can we split the logic so the handlers update state informing how to filter then call a shared filter method? There are some other dimensions we might filter on in the future, so duplicating all the logic everywhere might get unwieldy.

File: src/components/TopIssues.js [hunk 3]
```diff
@@ -67,9 +95,22 @@ class TopIssues extends React.Component {
         return (
             <div>
                 <div className="section-heading"><FontAwesomeIcon icon={faExclamationTriangle} size="lg" /><h2>Top Issues</h2></div>
-                <div className="mb-20">
-                    <input className="pointer" id="showmanual-checkbox" type="checkbox" onChange={(e) => { this.handleCheckChange(e) }} />
-                    <span className="pointer" onClick={(e) => { this.handleCheckClick(e) }}>Hide Manual Checks</span>
+                <div className=" mb-20 issue-select-bar">
+                    <div>
+                        <span>Select Issue Type:</span>
+                        <select className="ml-10 issue-dropdown" name="select-issue-type" id="select-issue-type" onChange={(e) =>{this.handleIssueSelect(e)}} >
```
Comment (Winerip) [inline src/components/TopIssues.js:101]: lint: line overrun. Would be more readable as:
```
                        <select className="ml-10 issue-dropdown"
                                name="select-issue-type"
                                id="select-issue-type"
                                onChange={(e) =>{this.handleIssueSelect(e)}} >
```

File: src/components/TopIssues.js (hunk 1 above)
Comment (rousik) [reply src/components/TopIssues.js:40]: Alternatively, single filter may work here:

```
this.state.data.filter(item =>
  item.category === issue
  && (this.state.showManual || item.manual == false))
```

File: src/components/TopIssues.js [hunk 4]
```diff
@@ -24,6 +26,23 @@ class TopIssues extends React.Component {
...
+            this.setState({            
+                filteredData: (this.state.showManual ? this.state.data.filter(item => item.category === issue) : this.state.data.filter(item => item.category === issue && item.manual === false)),
+                selectedIssue: issue          
+            });           
+        }
+    }
+
     handleCheckClick(e) {
         this.handleCheckChange(e);
         let cb = document.getElementById("showmanual-checkbox");
```
Comment (rousik) [inline src/components/TopIssues.js:48]: The following can be turned to one-liner: `cb.checked = !cb.checked`

File: src/components/TopIssues.js [hunk 5]
```diff
@@ -67,9 +95,22 @@ class TopIssues extends React.Component {
...
+                            <option value="all-issues">All Issues</option>
+                            <option value="performance">Performance</option>  
+                            <option value="accessibility">Accessibility</option>                                              
+                            <option value="best-practices">Best Practices</option>
+                            <option value="seo">SEO</option>
+                            <option value="pwa">Progressive Web App</option>
+                        </select>                    
+                    </div>
+                    <div>
+                        <input className="pointer" id="showmanual-checkbox" type="checkbox" onChange={(e) => { this.handleCheckChange(e) }} />
```
Comment (rousik) [inline src/components/TopIssues.js:111]: I would prefer more descriptive name for the method. `handleCheckChange` doesn't tell you that what is being checked/clicked is related to showing/hiding manual issues. I would probably try to name it something along the lines of `changeVisibilityOfManualIssues`.

It seems that you could collapse `handleCheckChange` and `handleCheckClick` into a single method that will determine what needs to happen based on `event.type` value (either click or change), [see here for details](https://reactjs.org/docs/events.html). 

Also, if I'm not mistaken, the event handlers will get event passed by default, so `onChange={(e) => {this.handleCheckChange(e)}` may be equivalent to simpler `onChange={this.handleCheckChange}`.

File: src/components/TopIssues.js (hunk 1 above)
Comment (suryalijo) [reply src/components/TopIssues.js]: Single filter works perfectly. Removed the long ternary line.

File: src/components/TopIssues.js (hunk 2 above)
Comment (suryalijo) [reply src/components/TopIssues.js]: Removed the condition check in the event handler and added code to call a shared filter method to update "filteredData". The selected issue type and the showManual state is passed as arguments.

File: src/components/TopIssues.js (hunk 3 above)
Comment (suryalijo) [reply src/components/TopIssues.js]: Formatted attributes to improve code readability.

File: src/components/TopIssues.js (hunk 4 above)
Comment (suryalijo) [reply src/components/TopIssues.js]: Replaced the condition check with this straightforward one-liner.

File: src/components/TopIssues.js (hunk 5 above)
Comment (suryalijo) [reply src/components/TopIssues.js]: Renamed handleCheckChange to handleManualCheckVisibility which will be the event handler for both click and change events. Removed arrow functions and the explicit passing of e to the  handlers.

**Comment by suryalijo** [comment]
Demo at https://suryalijo.github.io/code4good-accessibility/

**Comment by Winerip** [comment]
Sorry, I didn't realize I had to specifically approve the changes after I requested them. I saw the update and saw Jan's approval, and thought that was enough.

File: src/components/TopIssues.js [hunk 6]
```diff
@@ -13,74 +12,63 @@ class TopIssues extends React.Component {
...
+
     handleClick() {
         if (!this.state.showMore) {
             this.setState({
                 showMore: true
             })
         }
     }
+
+    issueFilter(issue, showManual){
```
Comment (Winerip) [inline src/components/TopIssues.js:56]: nit: issueFilter is a noun-phrase, filterIssue would be a verb-phrase. All other methods are verb-phrases, and a verb-phrase is what I would expect for a function. Not really important, but just something to keep in mind when naming functions.

Comment (suryalijo) [reply src/components/TopIssues.js:56]: Sure, will keep that in mind.

File: src/components/TopIssues.js [hunk 7]
```diff
@@ -3,56 +3,72 @@ import DescriptionPopover from './DescriptionPopover.js';
...
         this.setState({
             data: arr,
             filteredData: arr
         });
+    }    
+
+    handleIssueSelect(e){        
+        let issue = e.target.value;  
+        this.setState({
+            filteredData : this.issueFilter(issue, this.state.showManual),
```
[the remark this answers is not in the document]
Comment (suryalijo) [reply src/components/TopIssues.js:32]: Yes, it will definitely improve the code.. I will make these changes subsequently and create a separate PR.

**Comment by suryalijo** [comment]
I can merge it.

**Pull request merged**